editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2097,8 +2098,14 @@ bool CanvasItemEditor::_gui_input_move(const Ref<InputEvent> &p_event) {
 		}
 
 		// Confirm the move (only if it was moved)
-		if (b.is_valid() && !b->is_pressed() && b->get_button_index() == MouseButton::LEFT) {
-			if (transform.affine_inverse().xform(b->get_position()) != drag_from) {
+		if ((b.is_valid() && b->get_button_index() == MouseButton::LEFT && !b->is_pressed()) || tool_changed) {
+			bool was_dragged = false;
+			if (b.is_valid() && b->get_button_index() == MouseButton::LEFT && !b->is_pressed()) {
+				was_dragged = transform.affine_inverse().xform(b->get_position()) != drag_from;
+			} else{
+				was_dragged = transform.affine_inverse().xform(m->get_position()) != drag_from;
```

Comment:
This is checked in an if condition upstream that this code is nested in after the comment // Move the nodes

Reply:
No that is another if statement, it doesn't cover this

Reply:
You have:
```cpp
if (!tool_changed && m.is_valid() && !drag_selection.is_empty()) {
    ...
}

if (tool_changed || (b.is_valid() && !b->is_pressed() && b->get_button_index() == MouseButton::LEFT)) {
    // Nothing says `m` is valid here, it could be either or
}

Reply:
You're correct, I was confused by me fat fingering the previous if statements and getting confused by the code scopes.

Reply:
This still needs to be fixed, this will crash if `m` is not valid, which it isn't guaranteed to be if `(b.is_valid() && !b->is_pressed() && b->get_button_index() == MouseButton::LEFT)` is false, for example it enters this part if just `b->get_button_index() != MouseButton::LEFT` for example, I'm not sure what this different solution is designed to do

Reply:
Hmm, I'm not seeing where it would crash. If it enters with those conditions and m isn't valid, was_dragged never gets set to true, and no changes are committed.

Reply:
Oh yeah, that was resolved in my last push.

Reply:
Explain how you solved that, where do you make sure that `m` is valid?

Reply:
My bad missed that line 🙂

Reply:
Isn't this check sufficient? 

![image](https://github.com/godotengine/godot/assets/105675984/6748758e-259f-44f5-b692-5b4ecb72c27f)

Reply:
It is 🙂, missed that line, so much to review

Reply:
> My bad missed that line 🙂

All good

Reply:
> It is 🙂, missed that line, so much to review

You're lightning-quick haha, keep it up. 

Thanks a lot for your help.